tests/test_funcs.py

```diff
@@ -1637,3 +1643,157 @@ def test_kind(self, xp: ModuleType, library: Backend):
+                result = f(
+                    *(
+                        (array[index] if array is not None else None)
+                        for array in arrays
+                    ),
+                    *other_args,
+                    **kwargs,
+                )
+                # Assume `result` is either a tuple or single array. This is easily
+                # generalized by allowing the contributor to pass an `unpack_result`
+                # callable to the decorator factory.
+                result = (result,) if not isinstance(result, tuple) else result
+                results.append(result)
+            results = list(zip(*results, strict=True))
+
+            # Reshape results
+            for i, result in enumerate(results):
+                result = xp.stack(result)  # noqa: PLW2901
+                core_shape = result.shape[1:]
+                results[i] = xp.reshape(result, batch_shape + core_shape)
+
+            # Assume `result` should be a single array if there is only one element or
+            # a `tuple` otherwise. This is easily generalized by allowing the
+            # contributor to pass an `pack_result` callable to the decorator factory.
+            return results[0] if len(results) == 1 else results
+
+        return wrapper
+
+    return decorator
+
+
+@_apply_over_batch(("a", 1), ("v", 1))
+def xp_searchsorted(a, v, side, xp):
+    return xp.searchsorted(xp.asarray(a), xp.asarray(v), side=side)
+
+
+@pytest.mark.skip_xp_backend(Backend.DASK, reason="no take_along_axis")
+@pytest.mark.skip_xp_backend(Backend.SPARSE, reason="no searchsorted")
+class TestSearchsorted:
+    @pytest.mark.parametrize("side", ["left", "right"])
+    @pytest.mark.parametrize("ties", [False, True])
+    @pytest.mark.parametrize(
+        "shape", [0, 1, 2, 10, 11, 1000, 10001, (2, 0), (0, 2), (2, 10), (2, 3, 11)]
+    )
+    @pytest.mark.parametrize("nans_x", [False, True])
+    @pytest.mark.parametrize("infs_x", [False, True])
+    def test_nd(self, side, ties, shape, nans_x, infs_x, xp):
+        if nans_x and is_torch_namespace(xp):
+            pytest.skip("torch sorts NaNs differently")
+        rng = np.random.default_rng(945298725498274853)
+        x = rng.integers(5, size=shape) if ties else rng.random(shape)
+        # float32 is to accommodate JAX - nextafter with `float64` is too small?
+        x = np.asarray(x, dtype=np.float32)
+        xr = np.nextafter(x, np.inf)
+        xl = np.nextafter(x, -np.inf)
+        x_ = np.asarray([-np.inf, np.inf, np.nan])
+        x_ = np.broadcast_to(x_, (*x.shape[:-1], 3))
+        y = rng.permuted(np.concatenate((xl, x, xr, x_), axis=-1), axis=-1)
+        if nans_x:
+            mask = rng.random(shape) < 0.1
+            x[mask] = np.nan
+        if infs_x:
+            mask = rng.random(shape) < 0.1
+            x[mask] = -np.inf
+            mask = rng.random(shape) > 0.9
+            x[mask] = np.inf
+        x = np.sort(x, stable=True, axis=-1)
```

Comment:
This is unnecessary and causing CI failures.
```suggestion
        x = np.sort(x, axis=-1)
```